### ISSUE TITLE
Duplicate messages on pubsub channel after redis restart

### ISSUE THREAD

**Issue by oferitz** [capriza/node-busmq#48]
Hi, i'm experiencing strange behaviour when using pubsub.
I have 2 node.js applications app1 is publishing to a channel and app2 subscribed to this channel.
Both apps and Redis itself are Dockerized and both of the apps using the ioredis driver.

```
const Bus = require('busmq')
const bus = Bus.create({driver: 'ioredis', 'redis://localhost:6379'})
const pubsub = bus.pubsub('myChannel')

bus.on('error', (err) => {
  console.log('(AN ERROR HAS OCCURRED)',  err)
})

bus.on('online', () => {
  console.log('(REDIS IS ONLINE)')
  pubsub.on('message', (message) => {
      console.log('(***GOT NEW MESSAGE***)', message)
  })

  if (!pubsub.isSubscribed()) {
    pubsub.subscribe()
  }
})

bus.on('offline', () => {
  logger.info('(REDIS IS DOWN...)')
})

bus.connect()
```
The problem start when i restart the Redis container, each restart is causing the `message` event to be called multiple times with the same message.

At first i thought that what causing this is the fact that each restart is emitting the `online` event which then register the `message` event again and again in each restart which is actually what it does, i inspected the `pubsub._events.message` and saw that indeed after 4 restarts it registered the `message` event 4 times! more than that the messages were duplicated by 4 each restart!
so i tried the following hack just to verify that my assumptions were true:

```
bus.on('online', () => {
  if (!pubsub._events.message) {
    pubsub.on('message', (message) => {
      console.log('(***GOT NEW MESSAGE***)')
    })
  }

  if (!pubsub.isSubscribed()) {
    pubsub.subscribe()
  }
}) 
```
but that didn't help... it only decreased the number of duplicate messages to be linear and indeed after doing this the `message` event is registered only once but still after each restart the duplicate messages are increasing by 1 for each restart.

I found this https://github.com/ortoo/node-redis-sentinel/issues/31 which seems similar to my problem but i'm not sure.

Any ideas?

**Comment by fujifish** [comment]
I'll try to reproduce this locally

**Comment by fujifish** [comment]
Can you share the publisher code?

**Comment by oferitz** [comment]
I actually tested also the publisher code and it doesn't publish duplicate events but here it is:

```
  const publishEvent = async function (msg) => {
    if (bus.isOnline()) {
      try {
        await pubsub.publish(msg)
        return Promise.resolve('event was published')
      } catch (e) {
        return Promise.reject(e)
      }
    } else {
      logger.info('(COULD NOT PUBLISH EVENT - EVENT BUS IS OFFLINE)')
      throw new Error('event was not published. redis is probably offline')
    }
  }
}
```

**Comment by fujifish** [comment]
I've managed to reproduce, working on a fix

**Comment by oferitz** [comment]
@fujifish thanks for the quick response and action - highly appreciated.

**Comment by fujifish** [comment]
released version 2.0.1 with the fix
